### PR TITLE
Manage vocab page

### LINKED ISSUE #233: Create a Manage Vocab Items page
**Goals**
Screen Shot 2022-10-13 at 12 27 54 AM

Screen Shot 2022-10-13 at 12 27 02 AM

Create a `ManageVocab` page that has the same functionality as `ManageLessons` and `ManageUnits`:
1. Selecting and unselecting vocab items
2. Deleting vocab items
3. Reordering vocab items

**Steps**

Client:
1. Duplicate the functionality from the `ManageLessons` and `ManageUnits` pages. You'll have to create a new `ManageVocab` page.
2. Add the necessary functionality to `LessonHome` to only show selected vocab items.

API:
1. We'll have to add a `selected` field to the VocabItem Mongoose Schema in `api/src/models/language.js`. This selected field will be used on the `ManageVocab` page. Also make sure to update the vocab endpoints with this new change if necessary.

**Extra**
There will have to be a good amount of testing done with this issue. Make sure to consider all edge cases and mess around with selecting/unselected vocab items and seeing if the changes are reflected on the `LessonHome` page.

### PR DESCRIPTION
**Pull request by jamieRollison**
## Status:
:rocket: Ready

## Description
Adds a manage vocab page in the style of the manage lessons and manage units page. Updates the backend and redux to have the capacity to manage bulk vocab item updates. Adds the selected field to vocab items. Adds selected fields to example data in the test suites.

This was a very full stack issue and I definitely may have done it sub-optimally; looking forward to code review!

Fixes #233 

## Todos

Hitting the save button feels slow but this is the case for other "manage" pages, so there may be a way to optimize all of them

### CODE REVIEW

File: api/src/api/language/vocab.js [hunk 1]
```diff
@@ -103,4 +103,56 @@ router.post(
...
+      const vocabIndex = getVocabIndexByID(vocab_updates[i]._id, lesson);
+
+      if (vocabIndex === NOT_FOUND_INDEX) {
+        return sendResponse(res, 404, 'Vocab item not found');
+      }
+
+      /* Using the index, apply changes to the lesson data */
+      let vocabData = lesson.vocab[vocabIndex];
+
+      for (var key in vocab_updates[i]) {
```
Comment (ashayp22) [inline api/src/api/language/vocab.js:140]: I think there exists `patchDocument` helper method with the same functionality as this piece of code. The method is in `api/src/utils/languageHelper.`

File: api/src/models/language.js [hunk 2]
```diff
@@ -38,6 +38,7 @@ const Vocab = new mongoose.Schema({
   translation: { type: String, required: true },
   image: { type: String, required: false, default: '' },
   audio: { type: String, required: false, default: '' },
+  selected: { type: Boolean, required: true, default: false },
```
Comment (ashayp22) [inline api/src/models/language.js:41]: nice

File: api/test/mock-data/non-latin-mock-data.js [hunk 3]
```diff
@@ -88,6 +88,7 @@ module.exports.POST_BERBER_VOCAB_ITEM_EXPECTED = {
   audio: '',
   image: '',
   notes: '',
```
Comment (ashayp22) [inline api/test/mock-data/non-latin-mock-data.js:90]: Nice job updating all of the unit tests.

File: client/src/pages/ManageVocab/ManageVocab.js [hunk 4]
```diff
@@ -0,0 +1,135 @@
...
+    let selectedList = []
+    let unselectedList = []
+    for (let i = 0; i < lessonData.vocab.length; i += 1) {
+      const item = lessonData.vocab[i]
+
+      const formattedItem = {
+        _id: item._id,
+        title: item.original,
+        body: item.translation,
+        indicatorType: INDICATOR_TYPES.NONE, // TODO: remove hard-coded value
```
Comment (ashayp22) [inline client/src/pages/ManageVocab/ManageVocab.js:35]: I think we can remove the TODO. We had indicators in last semester's designs, but we've moved away from them this year.

**Review by ashayp22 — CHANGES_REQUESTED** [review]
Great PR Jamie! Did you do some manual testing with this page by selecting and re-ordering a bunch of vocab items, and then checking if the change was properly reflected on the Lesson Home page?

File: api/src/api/language/vocab.js (hunk 1 above)
Comment (jamieRollison) [reply api/src/api/language/vocab.js]: I changed it to use `patchDocument` instead

**Review by ashayp22 — APPROVED** [review]

**Pull request merged**